bodhi/server/push.py

```diff
@@ -108,6 +112,17 @@ def push(username, cert_prefix, config, **kwargs):
                                    update.title)
 
                 # Skip unsigned updates (this checks that all builds in the update are signed)
+                if not update.signed and kwargs['check_sig_status']:
+                    # Only initialize the buildsys instance once
+                    if koji is None:
+                        koji = buildsys.get_session()
+                    for build in update.builds:
+                        if not build.signed:
+                            build_tags = build.get_tags(koji)
+                            if not update.release.pending_signing_tag in build_tags:
+                                click.echo('Update %s was refreshed as signed' % build.nvr)
+                                build.signed = True
```

Comment:
This function is pretty large as it is and should be refactored to use sub functions so it's cleaner. Can we make the above code block be a helper function instead? We can refactor the rest of this method as a separate effort.